CrowServe/CrowServe.h

```diff
@@ -114,25 +121,47 @@ class Socket {
                         if (bytes_received < 1) {
                             // TODO: error handling
                             std::cout << "error";
-                            return;
+                            break;
                         }
                         message = std::span<const char>{buffer, (u32)bytes_received};
                     }
 
                     switch (ProtocolType(main_header.protocol)){
                         case ProtocolType::ProtocolCrownLink: {
-                            m_crownlink_protocol.handle(message_header.message_type, message, handler);
+                            m_crownlink_protocol.handle(CrownLink::MessageType(message_header.message_type), message, handler);
                         } break;
                         case ProtocolType::ProtocolP2P: {
-                            m_p2p_protocol.handle(message_header.message_type, message, handler);
+                            //m_p2p_protocol.handle(P2P::MessageType(message_header.message_type), message, handler);
                         } break;
                     }
                 }
             }
         }};
     }
 
-    bool receive();
+template<typename T>
+void send_messages(ProtocolType protocol, T& message) {
+    if (m_state != SocketState::Ready) {
+        // TODO logging
+        return;
+    }
+
+    std::lock_guard{m_mutex};
+
+    Header header{"", 1, (u16)protocol, 1};
```

Comment:
If it's not a constructor it's more robust to use designated initalizer or zero init aith explicit assignments.